src/main/java/com/dao/DishDAO.java

```diff
@@ -0,0 +1,128 @@
+package com.dao;
+
+import com.model.Dish;
+
+import javax.sql.DataSource;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class DishDAO {
+
+    private static final String UPDATE_QUERY = "UPDATE Dishes SET dish=?, dish_type_id=?, price=? where id=?";
+    private static final String INSERT_QUERY = "INSERT INTO Dishes(id, dish, dish_type_id, price)  values (?,?,?,?)";
+    private static final String SELECT_QUERY = "SELECT * FROM Dishes WHERE id = ?";
+    private static final String DELETE_QUERY = "DELETE FROM Dishes WHERE id=?";
+    private static final String SELECT_ALL_QUERY = "SELECT * FROM Dishes";
+
+    private DataSource dataSource;
+    private static DishDAO instance;
+
+    private DishDAO() {
+    }
+
+    public static DishDAO getInstance() {
+        if (instance == null) {
+            instance = new DishDAO();
+        }
+        return instance;
+    }
+
+    public void setDataSource(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    public boolean create(Dish dish) {
+        boolean isCreated = false;
+        try (Connection connection = dataSource.getConnection()) {
+
+            final PreparedStatement sql = connection.prepareStatement(INSERT_QUERY);
+            sql.setLong(1, dish.getId());
+            sql.setString(2, dish.getDish().toString());
+            sql.setLong(3, dish.getDishTypeId());
+            sql.setLong(4, dish.getPrice());
+            sql.executeUpdate();
+            isCreated = true;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return isCreated;
+    }
+
+    public List<Dish> getAll() {
+        List<Dish> res = new ArrayList<>();
+        Statement statement;
+        try (Connection connection = dataSource.getConnection()) {
+            statement = connection.createStatement();
+
+            final ResultSet rs = statement.executeQuery(SELECT_ALL_QUERY);
+            while (rs.next()) {
+                res.add(createDish(rs));
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return res;
+    }
+
+    private Dish createDish(ResultSet rs) throws SQLException {
```

Comment:
I think it would be better to change name of this method to smth like parseDish or createDishEntity, because createDish sounds quite ambiguous

Reply:
done